Fetching

jess2187 opened the pull request:
Able to fetch data from /recommend/username

Need to make it display error messages. Currently, there is a placeholder.

<img width="1123" alt="image" src="https://user-images.githubusercontent.com/20562379/80420591-37c0a500-8898-11ea-920c-1453b62b8d05.png">

evanram commented on webpage/src/App.js:
```diff
@@ -1,59 +1,91 @@
-function Search() {
+
+const Search = () => {
+  const [query, setQuery] = useState('jess2187')
```
What is this

evanram replied:
Default should be blank, and auto-focused with a placeholder that hints what the user should do

evanram commented on webpage/src/App.css:
```diff
@@ -35,23 +25,19 @@
-    height: 27px;
-    margin-top: 8px;
-    font-size: 14px;
+  border: 1px solid #EAEAEA;
```
Use lowercase

evanram replied:
Use lowercase hex numbers

evanram commented on webpage/src/App.js:
```diff
@@ -1,59 +1,91 @@
+const Recommendations = (props) => {
+  return (
+    props.recs && props.recs.recommendations.map((item, i) => (
+      <Recommendation key ={i} subreddit={item.subreddit} confidence={item.confidence} />
```
remove extra space after key

evanram replied:
Extra space 
```suggestion
      <Recommendation key={i} subreddit={item.subreddit} confidence={item.confidence} />
```

evanram commented on webpage/src/App.js:
```diff
@@ -1,59 +1,91 @@
     <div className="recommendation">
-      <h3>{"#" + props.score + ":"}</h3>
-      <h3 style={{paddingLeft:"10px"}}>{"r/" + props.subreddit}</h3>
+      <h3>{"r/" + props.subreddit}</h3>
```
Need space between `r/Subreddit` and `Confidence: #`

evanram commented on webpage/src/App.js:
```diff
@@ -1,59 +1,91 @@
-
-  let recommendationComponents = recommendations.map((item, i) => 
-    <Recommendation key={i} {...item} />
+const Error = (props) => {
```
```suggestion
const Error = () => (
  <div>
    <h3>Something went wrong :(</h3>
  </div>
)
```

evanram replied:
Dont need props and also this is a single expression so it can omit 'return'

evanram commented on webpage/src/App.css:
```diff
@@ -70,16 +56,12 @@
 
 .recommendation {
```
Why are `.okButton` and `.recommendation` different styles than the rest?

evanram commented on webpage/src/App.js:
```diff
@@ -1,59 +1,91 @@
-function Search() {
+
+const Search = () => {
+  const [query, setQuery] = useState('jess2187')
```
Default search-bar should be blank, and auto-focused with a placeholder that hints what the user should do.

I don't think we need any component-local state here (`useState()`), since username is just stored in redux anyways.

evanram replied:
Oh also so we can prevent the red-lines in a username (see ur picture), you can disable spellcheck.

evanram commented on webpage/src/App.js:
```diff
@@ -1,59 +1,91 @@
+            className="SearchInput"
+            type="text"
+            value = {query}
+            placeholder="Username"
```
Idea: placeholder should be a call-to-action that hints to user what they should enter in. Maybe "Username" is good though, idk.

evanram commented on webpage/src/App.js:
```diff
@@ -1,59 +1,91 @@
     <div className="recommendation">
-      <h3>{"#" + props.score + ":"}</h3>
-      <h3 style={{paddingLeft:"10px"}}>{"r/" + props.subreddit}</h3>
+      <h3>{"r/" + props.subreddit}</h3>
```
Need a space between these, b/c they render squashed together.

Ideally the subreddit and confidence shouldn't just be concatenated together, and we should use different styles to show the separation.

evanram replied:
Also use single quotes

evanram commented on webpage/src/App.js:
```diff
@@ -1,59 +1,91 @@
-
-  let recommendationComponents = recommendations.map((item, i) => 
-    <Recommendation key={i} {...item} />
+const Error = (props) => {
```
```suggestion
const ErrorPopup = () => (
  <div>
    <h3>Something  went wrong :(</h3>
  </div>
)
```

We can omit return since this is a single expr. Also we'll want to add the actual error message and maybe make this a Modal or Popup.

evanram commented on webpage/src/App.js:
```diff
@@ -1,59 +1,91 @@
-      {/* <img src='./loading.gif' className="App-logo" alt="logo" /> */}
-      {recommendationComponents}
+      
+      <div>  
```
Logic here is broken.

Here's what we need:
* On error: show error modal/popup, and then on close we reset to default state prompting to ask user
* On loading: gray out (disable) the search bar and show a loading gif
* On success: remove username from search bar or something that has good UX

evanram commented on webpage/src/actions.js:
```diff
@@ -0,0 +1,19 @@
+export const PERFORM_REQUEST = 'PERFORM_REQUEST'
```
2 space indents here

evanram commented on webpage/src/index.js:
```diff
@@ -1,17 +1,14 @@
 ReactDOM.render(
   <React.StrictMode>
+    <Provider store={store}>
     <App />
```
```suggestion
    <App/>
```

evanram commented on webpage/src/index.js:
```diff
@@ -1,17 +1,14 @@
-import './index.css';
 import App from './App';
-import * as serviceWorker from './serviceWorker';
+import store from './store';
```
```suggestion
import store from './store'
```

evanram replied:
and the other semicolons

evanram commented on webpage/src/recommendations.js:
```diff
@@ -0,0 +1,55 @@
+import axios from 'axios'
```
2 space indents not 4

evanram commented on webpage/src/store.js:
```diff
@@ -0,0 +1,44 @@
+import {createStore, applyMiddleware} from 'redux'
```
2 spaces

jess2187 commented:
![image](https://user-images.githubusercontent.com/20562379/80552000-864a6e00-8982-11ea-969f-5490c7f390a5.png)

![image](https://user-images.githubusercontent.com/20562379/80552180-0375e300-8983-11ea-8408-4cccb274bd28.png)

Corrected all the issues and finished the frontend.
- Added modal popup when there is a error
- Fixed the recommendation boxes UI
    - Made the subreddits links
- Added a loading gif
- Added a header to the recommendations

evanram commented:
nice
